halo/attest/keeper/cpayload.go

```diff
@@ -8,22 +8,27 @@ import (
 	"github.com/omni-network/omni/halo/attest/types"
 	evmenginetypes "github.com/omni-network/omni/halo/evmengine/types"
 	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/xchain"
 
 	abci "github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	"github.com/cosmos/gogoproto/proto"
 )
 
-var _ evmenginetypes.CPayloadProvider = CPayloadProvider{}
+var _ evmenginetypes.CPayloadProvider = Keeper{}
 
-// CPayloadProvider implements the CPayloadProvider interface for the attester module.
-// It extracts the aggregate attestations from the vote extensions of the last local commit.
-type CPayloadProvider struct {
-}
+func (k Keeper) PreparePayload(ctx context.Context, height uint64, commit abci.ExtendedCommitInfo) ([]sdk.Msg, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	if err := baseapp.ValidateVoteExtensions(sdkCtx, k.skeeper, int64(height), sdkCtx.ChainID(), commit); err != nil {
+		log.Error(ctx, "Cannot include invalid vote extensions in payload", err, "height", height)
+		return nil, nil
```

Comment:
Is it intentional that we log an error, however, we return `nil` for the `error` return value? I find this a little confusing.